### ISSUE TITLE
Nullable type hint on a property hook parameter is tokenized as T_INLINE_THEN (ternary), breaking Squiz.WhiteSpace.OperatorSpacing

### ISSUE THREAD

**Issue by TonnyJe** [PHPCSStandards/PHP_CodeSniffer#1494]
## Describe the bug

The `?` of a nullable type hint on a property hook accessor parameter
(`set(?int $value)`) is tokenized as `T_INLINE_THEN` (the ternary operator)
instead of `T_NULLABLE`. The identical parameter on a regular method is
tokenized correctly as `T_NULLABLE`.

As a result `Squiz.WhiteSpace.OperatorSpacing` treats the `?` as a ternary
operator and demands surrounding whitespace. `phpcbf` "fixes" it to
`set( ? int $value)`, which is invalid formatting.

`phpcs -vv` confirms the misclassification:

```text
* token 18 changed from ? to T_INLINE_THEN   (the ? in set(?int $value))
* token 49 changed from ? to T_NULLABLE      (the ? in function setBaz(?int $value))
```

### Code sample
```php
<?php

class Foo
{
    public int $bar {
        set(?int $value) {
            $this->bar = $value ?? 0;
        }
    }
}
```

### Custom ruleset

None required. Reproduced with the bundled `Squiz` standard and a single sniff:

```xml
<?xml version="1.0"?>
<ruleset name="repro">
  <rule ref="Squiz.WhiteSpace.OperatorSpacing"/>
</ruleset>
```

### To reproduce
1. Save the code sample as `test.php`.
2. Run `phpcs -s --standard=Squiz --sniffs=Squiz.WhiteSpace.OperatorSpacing test.php`

```text
FILE: /path/to/test.php
--------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
--------------------------------------------------------------------------------
 6 | ERROR | [x] Expected 1 space before "?"; 0 found
   |       |     (Squiz.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 6 | ERROR | [x] Expected 1 space after "?"; 0 found
   |       |     (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
--------------------------------------------------------------------------------
```

3. Run `phpcbf --standard=Squiz --sniffs=Squiz.WhiteSpace.OperatorSpacing test.php`. The accessor becomes:

```php
        set( ? int $value) {
```

### Contrast (not flagged)

The same nullable parameter on a regular method tokenizes as `T_NULLABLE` and
produces no error:

```php
<?php

class Foo
{
    public function setBaz(?int $value): void
    {
        $this->baz = $value ?? 0;
    }
}
```

## Expected behavior

The `?` in `set(?int $value)` should be tokenized as `T_NULLABLE`, exactly as it
is in a regular method parameter list, and `Squiz.WhiteSpace.OperatorSpacing`
should report nothing. `phpcbf` should leave the file unchanged.

## Versions (please complete the following information)

|                         |                                                        |
|-------------------------|--------------------------------------------------------|
| Operating System        | Alpine Linux v3.24 (official `composer:2` Docker image) |
| PHP version             | 8.5.9                                                   |
| PHP_CodeSniffer version | 4.0.4; also reproduced on the `4.x` branch (4.0.5-dev)  |
| Standard                | Squiz (single sniff `Squiz.WhiteSpace.OperatorSpacing`) |
| Install type            | Composer (local)                                        |

## Additional context

- Root cause appears to be that the tokenizer's `T_NULLABLE` vs `T_INLINE_THEN`
  disambiguation does not recognise a property hook accessor's parameter list as
  a parameter-list context.
- Related: #1493 (`Generic.WhiteSpace.ScopeIndent` on property hook bodies). Both
  seem to stem from the tokenizer not modelling property hook accessors, but the
  affected code paths and sniffs are different.

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.
